docs/articles/advanced_topics/generic_host.md

```diff
@@ -0,0 +1,142 @@
+            TokenType = TokenType.Bot,
+            Intents = DiscordIntents.AllUnprivileged
+        });
+    }
+
+    public async Task StartAsync(CancellationToken token)
+    {
+        await _discordClient.ConnectAsync();
+        // Other startup things here
+    }
+
+    public async Task StopAsync(CancellationToken token)
+    {
+        await _discordClient.DisconnectAsync();
+        // More cleanup possibly here
+    }
+}
+```
+>[!WARNING]
+> Hard-coding your bot token into your source code is not a good idea, especially if you plan to distribute your code publicly. This is because anyone with access to your code can easily extract your bot token, which can be used to perform unauthorized actions on your bot account.
+>Instead, it's recommended that you store your bot token in a secure location, such as a configuration file, environment variable, or secret storage service. You can then retrieve the token at runtime and pass it to the initializer. See for example [How to use a configuration with GEneric Host](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-7.0) or [how to use an environment variable](https://learn.microsoft.com/en-us/dotnet/api/system.environment.getenvironmentvariable?view=net-7.0). 
+
+The `StartAsync()` method contains the code that runs when your application starts. In this case, the `DiscordClient` connects to the Discord API.
+
+The `StopAsync()` method contains the code that runs when your application is shut down. In the case of a bot, this might involve closing connections to external APIs, releasing resources, or performing other cleanup tasks.
+
+By implementing these methods, you can ensure that your application starts and stops cleanly, and that any necessary resources are properly managed. This makes it easier to build robust, reliable applications that can be run as services.
+
+With this class, you can easily create and run your own bot services using the .NET Generic Host. Just replace the Token property with your own Discord bot token, and you're ready to go!
+
+## Using Serilog with the Generic Host
+Logging is an important part of any application, especially one that runs as a service. Fortunately, the .NET Generic Host makes it easy to integrate with popular logging libraries, like Serilog.
+
+### Dependencies
+You will need the [`Serilog.Extensions.Hosting`](https://www.nuget.org/packages/Serilog.Extensions.Hosting) package (along with [`Serilog`](https://www.nuget.org/packages/Serilog) itself with whichever sinks you prefer), which are available from NuGet.
+
+### In your Service
+Then, you will need to add Serilog to your DiscordBot initializer block to ensure that your bot logs messages using Serilog. To do this, you will need to create a new `LoggerFactory` object and add the Serilog logger provider to it. You can also specify a minimum log level and silence certain DSharpPlus events, such as the "unknown event" log.
```

Comment:
nitpick: this initializer block doesn't initialize DiscordBot, but DiscordClientConfiguration